Source/WebKit2/NetworkProcess/soup/NetworkProcessSoup.cpp

```diff
@@ -50,9 +52,12 @@ void NetworkProcess::userPreferredLanguagesChanged(const Vector<String>& languag
     SoupNetworkSession::defaultSession().setAcceptLanguages(languages);
 }
 
-void NetworkProcess::setProxies(WebCore::SessionID, const Vector<WebCore::Proxy>& proxies)
+void NetworkProcess::setProxies(WebCore::SessionID sessionID, const Vector<WebCore::Proxy>& proxies)
 {
-    SoupNetworkSession::defaultSession().setProxies(proxies);
+    if (auto *storageSession = NetworkStorageSession::storageSession(sessionID))
```

Comment:
This change  ensures that on private browser sessions the proxy's are set for each session is separate.